### PR TITLE
Add minor doc strings to some internal functions

### PR DESCRIPTION
**Pull request by wschin**
This PR adds some minor doc strings for devs who want to know more about matrix factorization.

### CODE REVIEW

File: src/Microsoft.ML.Recommender/SafeTrainingAndModelBuffer.cs [hunk 1]
```diff
@@ -275,6 +275,9 @@ private unsafe void Dispose(bool disposing)
                     }
                     rowGetter(ref row);
                     // REVIEW: Instead of ignoring, should I throw in the row > rowCount case?
+                    // The index system in the LIBMF (the underlying library trains the model) is 0-based, so we need
+                    // to minus one from 1-based indexes returned by ML.NET's key-valued getters. We also skip 0 returned
```
Comment (shauheen) [inline src/Microsoft.ML.Recommender/SafeTrainingAndModelBuffer.cs]: >minus [](start = 26, length = 5)

deduct

File: src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs [hunk 2]
```diff
@@ -255,6 +255,16 @@ private ValueGetter<float> GetGetter(ValueGetter<uint> matrixColumnIndexGetter,
             return mapper as ValueMapper<TMatrixColumnIndexIn, TMatrixRowIndexIn, TOut>;
         }
 
+        /// <summary>
+        /// Compute the (approximated) value at the <paramref name="srcCol"/>-th column and the
+        /// <paramref name="srcRow"/>-th row. Notice that both of <paramref name="srcCol"/> and
+        /// <paramref name="srcRow"/> are 1-based indexes, so the first row/column index is 1.
+        /// The reason of having 1-based indexing system is that key-valued getter in ML.NET returns
```
Comment (shauheen) [inline src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs]: >of [](start = 23, length = 2)

for

File: src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs [hunk 3]
```diff
@@ -267,9 +277,22 @@ private void MapperCore(in uint srcCol, ref uint srcRow, ref float dst)
                 dst = float.NaN;
                 return;
             }
+
+            // The index system in the LIBMF (the library trains the model) is 0-based, so we need to minus one
```
Comment (shauheen) [inline src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs]: >minus [](start = 102, length = 5)

deduct

File: src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs [hunk 4]
```diff
@@ -267,9 +277,22 @@ private void MapperCore(in uint srcCol, ref uint srcRow, ref float dst)
...
 
+        /// <summary>
+        /// Compute the (approximated) value at the <paramref name="columnIndex"/>-th column and the
+        /// <paramref name="rowIndex"/>-th row. Notice that, in contrast to <see cref="MapperCore"/>,
+        /// both of <paramref name="columnIndex"/> and <paramref name="rowIndex"/> are 0-based indexes,
+        /// so the first row/column index is 0.
+        /// </summary>
+        /// <param name="columnIndex">0-based column index.</param>
+        /// <param name="rowIndex">0-based row index.</param>
+        /// <returns></returns>
```
Comment (shmoradims) [inline src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs]: empty tags might cause warnings on the docs engine. please either fill it or remove it.

**Pull request merged**